pkg/asset/machines/aws/master.go

```diff
@@ -0,0 +1,77 @@
+// Package aws generates Machine objects for aws.
+package aws
+
+import (
+	"text/template"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// MasterConfig is used to generate master machines
+type MasterConfig struct {
+	ClusterName string
+	AMIID       string
+	Tags        map[string]string
+	Region      string
```

Comment:
I think populating this will be more straightforward if we just add an [`AWSPlatform`][1] property.

[1]: https://github.com/openshift/installer/blob/ace571b041049c276443631f349dbd9e0dad27e5/pkg/types/installconfig.go#L110

Reply:
That opens up the API to vpc and cidr block too, which might not be ideal.

Reply:
> You could even use DefaultMachinePlatform instead of your current Machine

that opens up the api to vpc, vpccidr block, which might not be ideal

> You could even use DefaultMachinePlatform instead of your current Machine.

I don't think that would be correct, we need to account for 3 sources to create the machine.
- installer defaults for unfilled values in install config
- defaultmachineplatform
- machinepoolplatform

Reply:
> that opens up the api to vpc, vpccidr block, which might not be ideal

So create a new type for the shared fields (region and tags, maybe also the AMI?)?  You could also push tags and the AMI down into `AWSMachinePoolPlatform`, although you'd want to leave a tags entry on `AWSPlatform` for tagging non-machine resources.  But I'd like to minimize the number of slightly-different phrasings of types in this space if possible.

> > You could even use `DefaultMachinePlatform` instead of your current `Machine`.
>
> I don't think that would be correct, we need to account for 3 sources to create the machine.
>
> * installer defaults for unfilled values in install config
> * defaultmachineplatform
> * machinepoolplatform

Sorry, I was trying to suggest using the `DefaultMachinePlatform` property on a local instance of the `AWSPlatform` type.  You could fill in that information using all of the usual install-config sources.  I agree you don't want to use the install-config `AWSPlatform.DefaultMachinePlatform` instance directly.